modules/swagger-codegen/src/main/resources/swift3/Models.mustache

```diff
@@ -202,6 +205,20 @@ class Decoders {
             }{{/isEnum}}
             {{^isEnum}}instance.{{name}} = Decoders.decodeOptional(clazz: {{{baseType}}}.self, source: sourceDictionary["{{baseName}}"] as AnyObject?){{/isEnum}}{{/allVars}}
             {{/unwrapRequired}}
+
+            {{#parent}}
+            for key in keys {
+                propsDictionary.removeValue(forKey: key)
+            }
+
+            for key in propsDictionary.keys {
+                if let decodedValue = Decoders.decodeOptional(clazz: String.self, source: propsDictionary[key] as AnyObject?) {
+                    instance[key as! String] = decodedValue
+                } else {
+                    instance[key as! String] = nil
```

Comment:
We wanted to keep propsDictionary the same as sourceDictionary (Its just a copy).  sourceDictionary uses AnyHashable.  So we need to cast to a string we could add more protection here but its not really needed.